ios-sdk/core-data.md

```diff
@@ -0,0 +1,180 @@
+  guard let appDelegate = UIApplication.shared.delegate as? AppDelegate
+  else {
+    return
+  }
+
+  // Step 2 create an instance of NSManagedObjectContext
+  let managedContext = appDelegate.persistentContainer.viewContext
+
+  // Step 3 create an instance of NSManagedObjectContext which
+  // just fetches all the attribute information for the given entity
+  // using NSEntityDescription and then creates a variable called
+  // repo which contains all the data required.
+
+  let entity = NSEntityDescription.entity(forEntityName: "Repo",
+                                          in: managedContext)!
+
+  let repo = NSManagedObject(entity: entity,
+                             insertInto: managedContext)
+  print("\(repo), REPO")
+
+  // Step 2 will return a Repo with name and star set to the default,
+  // (for me it will set it to nil for the name and 0 for the star,
+  // it could be different if you've set your default value to something different)
+  // We change this to the new ones that are being passed into this
+  // function in Step 3
+  task.setValue(name, forKeyPath: "name")
+  task.setValue(star, forKeyPath: "star")
+
+  // Step 4 we save the data using the .save() method. Please note
+  // that is recommended that you put this in a try catch block so that you
+  // don't crash the app if you run into an error when saving the data.
+  do {
+      try managedContext.save()
+  } catch let error as NSError {
+      print("Could not save. \(error), \(error.userInfo)")
+  }
+
+}
+```
+There you have it, if you run this function in the place where you are saving the data, it will now store this data in our database!
+
+### Remove data using Core Data Framework
+
+Now that we have the ability to save data, we can add another function that will allow us to delete the data.
+
+```js
+
+guard let appDelegate = UIApplication.shared.delegate as? AppDelegate
+else {
+    return
+}
+
+let managedContext = appDelegate.persistentContainer.viewContext
+
+repo.remove(at: indexPath.row)
+
+```
+
+You'll notice that our first couple of steps are very similar to when we are saving the data, Once we create an appDelegate and create a variable that store the context all we need to do is call `remove(at:Int)` where you enter the row number that you'd like to delete. For me as I am using a Table view I would write the above function as `remove(at:indexPath.row)`
+
+
+You've made it through to the end! you've successfully managed to use Core Data to add/remove & view data that is stored in your newly built database.
+
+Please remember that this was written during my transition from being a JavaScript developer to a Swift developer, in other words I'm a complete beginner to Swift and the iOS SDK, so if you find that I've missed something or that there is an alternative way achieve the above, I'd love to hear back from you and improve upon what already exists.
+
+#### **[Sohil Pandya](https://raw.githubusercontent.com/sohilpandya)**  
```

Comment:
also I'd recommend putting the bold inside the link:
`[**Sohil Pandya**](https://github.com/sohilpandya)`